src/components/CodeLineShapes/index.tsx

```diff
@@ -24,6 +24,7 @@ const CodeLineShapes = ({ lineShapes }: Props) => {
           <div
             key={[
               String(lineShapes.line),
+              String(shapeIndex),
```

Comment:
Why did you need to add this? Is it now possible to have more than one shape with the same `line`? Or is it because a `shape` doesn't necessarily have a `line` (which I don't fully understand, tbh)?

Reply:
There can be multiple shapes per line, yes. A shape is either whitespace or a code silhouette. That's all.